src/service_aggregator.py

```diff
@@ -7,186 +7,168 @@
-    #
-    #     for node in kg_nodes:
-    #         if 'name' in kg_nodes[node] and 'attributes' in kg_nodes[node]:
-    #             attribs = kg_nodes[node]['attributes']
-    #
-    #             for attrib in attribs:
-    #                 if "type" not in attrib:
-    #                     attrib['type'] = 'EDAM:data_0006'
-    #         else:
-    #             message['error'] = f'Error detected. Got kgraph node from strider missing the name or attributes ({node}), aborting.'
-    #             return message
-
+    strider_answer, status_code = strider(message)
+
+    # was any data returned
+    if len(strider_answer) == 0:
+        message['error'] = 'Strider error: Got empty response from strider.'
+        return message, status_code
+    # html error code returned, but at least there was something returned
+    elif status_code != 200:
+        logger.error(f'Strider error: HTML error status code {status_code} returned.')
+        strider_answer['error'] = f'Strider error: HTML error status code {status_code} returned.'
+        return strider_answer, status_code
+    # good html status code, but still do some checking
+    elif status_code == 200:
         logger.debug(f"strider in ({uid}): {json.dumps(message)}")
         logger.debug(f"strider out ({uid}): {json.dumps(strider_answer)}")
 
     # are we doing answer coalesce
     if coalesce_type != 'none':
         # get the request coalesced answer
-        coalesce_answer: dict = post('coalesce', f'https://answercoalesce.renci.org/1.1/coalesce/{coalesce_type}', strider_answer)
-
-        # was there an error getting data
-        if coalesce_answer is None:
-            logger.error("Error detected: Got no answer from Answer coalesce, aborting.")
-            message['error'] = 'Error detected: Answer coalesce failed to return an answer, aborting.'
-            return message
-        # did we get a good response
-        elif len(coalesce_answer) == 0:
-            logger.error("Error detected: Got an empty answer from Answer coalesce, aborting.")
-            message['error'] = 'Error detected: Got an empty answer from Answer coalesce, aborting.'
-            return message
-        else:
+        coalesce_answer, status_code = post('coalesce', f'https://answercoalesce.renci.org/1.1/coalesce/{coalesce_type}', strider_answer)
+
+        # was any data returned
+        if len(coalesce_answer) == 0:
+            strider_answer['error'] = 'Answer coalesce error: Got empty response from strider.'
+            return strider_answer, status_code
+        # html error code returned, but at least there was something returned
+        elif status_code != 200:
+            logger.error(f'Answer coalesce error: HTML error status code {status_code} returned.')
+            coalesce_answer['error'] = f'Answer coalesce error: HTML error status code {status_code} returned.'
+            return coalesce_answer, status_code
```

Comment:
If AC returns something other than a 200, but we had valid strider results, we should just pass the strider results to the next phase of the operation (omnicorp)